tsl/src/compression/compression.c

```diff
@@ -2740,37 +2755,29 @@ decompress_batches_for_insert(const ChunkInsertState *cis, TupleTableSlot *slot)
 	ScanKeyData *index_scankeys = build_index_scankeys_using_slot(cis->hypertable_relid,
 																  in_rel,
 																  out_rel,
-																  settings,
+																  key_columns,
 																  slot,
 																  &index_rel,
+																  &index_columns,
 																  &num_index_scankeys);
 
 	if (index_rel)
 	{
 		/*
-		 * Prepare the heap scan keys if any
-		 * This assumes that columns in segmentby are
-		 * handled by the index scan keys and potentially
-		 * might need to be handled by going through
-		 * index scan keys instead
+		 * Prepare the heap scan keys for all
+		 * key columns not found in the index
 		 */
-		Bitmapset *filtered_key_columns = NULL;
 		int i = -1;
-		while ((i = bms_next_member(key_columns, i)) > 0)
+		while ((i = bms_next_member(index_columns, i)) > 0)
 		{
-			AttrNumber attno = i + FirstLowInvalidHeapAttributeNumber;
-			char *attname = get_attname(out_rel->rd_id, attno, false);
-			if (!ts_array_is_member(settings->fd.segmentby, attname))
-			{
-				filtered_key_columns = bms_add_member(filtered_key_columns, i);
-			}
+			key_columns = bms_del_member(key_columns, i);
```

Comment:
Suspicious that codecov complains about this. Also `bms_difference` might be convenient here.